File: ServerSpace/src/main/java/Models/Tarifa/TarifaSaveModel.java
package Models.Tarifa;

import java.sql.Timestamp;

import com.fasterxml.jackson.annotation.JsonProperty;

import EntityLayer.TarifaEntity;
import Enumerados.ProcessActionEnum;

public class TarifaSaveModel {

    public TarifaSaveModel() {
        this.tarifaId = 0;
        this.objetoReferenciaId = 0;
        this.correlativo = "";
        this.monedaId = 0;
        this.tipoTarifaEnum = 0;
        this.unidadMedidaId = 0;
        this.costoTarifa = 0;
        this.nomTarifa = "";
        this.nomComercial = "";
        this.descripcion = "";
        this.fechaRegistro = null;
        this.codUsuario = "";
        this.estadoRegistro = false;
    }

    public TarifaSaveModel(TarifaEntity Ent) {
        this.tarifaId = Ent.getTarifaId();
        this.objetoReferenciaId = Ent.getObjetoReferenciaId();
        this.correlativo = Ent.getCorrelativo();
        this.monedaId = Ent.getMonedaId();
        this.tipoTarifaEnum = Ent.getTipoTarifaEnum();
        this.unidadMedidaId = Ent.getUnidadMedidaId();
        this.costoTarifa = Ent.getCostoTarifa();
        this.nomTarifa = Ent.getNomTarifa();
        this.nomComercial = Ent.getNomComercial();
        this.descripcion = Ent.getDescripcion();
        this.fechaRegistro = Ent.getFechaRegistro();
        this.codUsuario = Ent.getCodUsuario();
        this.estadoRegistro = Ent.getEstadoRegistro();
    }

    @JsonProperty("TarifaId")
    private int tarifaId = 0;

    public int getTarifaId() {
        return tarifaId;
    }

    public void setTarifaId(int tarifaId) {
        this.tarifaId = tarifaId;
    }

    @JsonProperty("ObjetoReferenciaId")
    private int objetoReferenciaId = 0;

    public int getObjetoReferenciaId() {
        return objetoReferenciaId;
    }

    public void setObjetoReferenciaId(int objetoReferenciaId) {
        this.objetoReferenciaId = objetoReferenciaId;
    }

    @JsonProperty("Correlativo")
    private String correlativo = "";

    public String getCorrelativo() {
        return correlativo;
    }

    public void setCorrelativo(String correlativo) {
        this.correlativo = correlativo;
    }

    @JsonProperty("NomTarifa")
    private String nomTarifa = "";

    public String getNomTarifa() {
        return nomTarifa;
    }

    public void setNomTarifa(String nomTarifa) {
        this.nomTarifa = nomTarifa;
    }

    @JsonProperty("NomComercial")
    private String nomComercial = "";

    public String getNomComercial() {
        return nomComercial;
    }

    public void setNomComercial(String nomComercial) {

        this.nomComercial = nomComercial;
    }

    @JsonProperty("Descripcion")
    private String descripcion = "";

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    @JsonProperty("FechaRegistro")
    private Timestamp fechaRegistro = null;

    public Timestamp getFechaRegistro() {
        return fechaRegistro;
    }

    public void setFechaRegistro(Timestamp fechaRegistro) {
        this.fechaRegistro = fechaRegistro;
    }

    @JsonProperty("CodUsuario")
    private String codUsuario = "";

    public String getCodUsuario() {
        return codUsuario;
    }

    public void setCodUsuario(String codUsuario) {
        this.codUsuario = codUsuario;
    }

    @JsonProperty("EstadoRegistro")
    private boolean estadoRegistro = false;

    public boolean getEstadoRegistro() {
        return estadoRegistro;
    }

    public void setEstadoRegistro(boolean estadoRegistro) {
        this.estadoRegistro = estadoRegistro;
    }

    @JsonProperty("TipoTarifaEnum")
    private int tipoTarifaEnum = 0;

    public int getTipoTarifaEnum() {
        return tipoTarifaEnum;
    }

    public void setTipoTarifaEnum(int tipoTarifaEnum) {
        this.tipoTarifaEnum = tipoTarifaEnum;
    }

    @JsonProperty("CostoTarifa")
    private double costoTarifa = 0;

    public double getCostoTarifa() {
        return costoTarifa;
    }

    public void setCostoTarifa(double costoTarifa) {
        this.costoTarifa = costoTarifa;
    }

    @JsonProperty("MonedaId")
    private int monedaId = 0;

    public int getMonedaId() {
        return monedaId;
    }

    public void setMonedaId(int monedaId) {
        this.monedaId = monedaId;
    }

    @JsonProperty("UnidadMedidaId")
    private int unidadMedidaId = 0;

    public int getUnidadMedidaId() {
        return unidadMedidaId;
    }

    public void setUnidadMedidaId(int unidadMedidaId) {
        this.unidadMedidaId = unidadMedidaId;
    }

    @JsonProperty("Action")
    private ProcessActionEnum action = ProcessActionEnum.Loaded;

    public int getAction() {
        return action.getValor();
    }

    public void setAction(ProcessActionEnum Action) {
        this.action = Action;
    }

}

File: SharedApi/Space/Space.Server/Model/Tarifa/TarifaSaveModel.cs
using Space.EntityLayer;
using System.Text.Json.Serialization;

namespace Space.Server.Model.Tarifa
{
    public class TarifaSaveModel
    {
        public TarifaSaveModel()
        {
            TarifaId = 0;
            ObjetoReferenciaId = 0;
            Correlativo = String.Empty;
            NomTarifa = String.Empty;
            NomComercial = String.Empty;
            Descripcion = String.Empty;
            FechaRegistro = DateTime.Now;
            CodUsuario = String.Empty;
            EstadoRegistro = true;
            TipoTarifaEnum = 0;
            CostoTarifa = 0;
            MonedaId = 0;
            UnidadMedidaId = 0;
            NomCompleto = String.Empty;
        }
        public TarifaSaveModel(TarifaEntity Item)
        {
            this.TarifaId = Item.TarifaId;
            this.ObjetoReferenciaId = Item.ObjetoReferenciaId;
            this.Correlativo = Item.Correlativo;
            this.NomTarifa = Item.NomTarifa;
            this.NomComercial = Item.NomComercial;
            this.Descripcion = Item.Descripcion;
            this.FechaRegistro = Item.FechaRegistro;
            this.CodUsuario = Item.CodUsuario;
            this.EstadoRegistro = Item.EstadoRegistro;
            this.TipoTarifaEnum = Item.TipoTarifaEnum;
            this.CostoTarifa = Item.CostoTarifa;
            this.MonedaId = Item.MonedaId;
            this.MonedaId = Item.MonedaId;
            this.UnidadMedidaId = Item.UnidadMedidaId;
            this.NomCompleto = Item.NomCompleto;
        }
        [JsonPropertyName("TarifaId")]
        public Int32 TarifaId { get; set; }
        [JsonPropertyName("ObjetoReferenciaId")]
        public Int32 ObjetoReferenciaId { get; set; }
        [JsonPropertyName("TipoTarifaEnum")]
        public Int16 TipoTarifaEnum { get; set; }
        [JsonPropertyName("CostoTarifa")]
        public Decimal CostoTarifa { get; set; }
        [JsonPropertyName("MonedaId")]
        public Int32 MonedaId { get; set; }
        [JsonPropertyName("UnidadMedidaId")]
        public Int32 UnidadMedidaId { get; set; }
        [JsonPropertyName("Correlativo")]
        public String Correlativo { get; set; }
        [JsonPropertyName("NomTarifa")]
        public String NomTarifa { get; set; }
        [JsonPropertyName("NomComercial")]
        public String NomComercial { get; set; }
        [JsonPropertyName("Descripcion")]
        public String Descripcion { get; set; }
        [JsonPropertyName("FechaRegistro")]
        public DateTime FechaRegistro { get; set; }
        [JsonPropertyName("CodUsuario")]
        public String CodUsuario { get; set; }
        [JsonPropertyName("EstadoRegistro")]
        public Boolean EstadoRegistro { get; set; }
        [JsonPropertyName("NomCompleto")]
        public string NomCompleto { get; set; }
    }
}
